Model/Connection.php

```diff
@@ -119,20 +122,40 @@ public function getProfile()
         return $this->profile;
     }
 
+    public function getConnection()
+    {
+        return $this->connection;
+    }
+
     /**
      * Execute a command
      *
-     * @param string $command
-     * @return string
+     * @param $command
+     * @return bool|string
+     * @throws ConnectionException
+     *
      */
     public function exec($command)
     {
         $this->assertConnected();
 
-        $stream = ssh2_exec($this->connection, $command);
-        stream_set_blocking($stream, true);
-        $streamOut = ssh2_fetch_stream($stream, SSH2_STREAM_STDIO);
-        return stream_get_contents($streamOut);
+        if( !($stream = ssh2_exec($this->connection, $command)) ) {
+            throw new ConnectionException(
+                sprintf('%s command failed through ssh', $command),
+                $this->profile
+            );
+        }
+        $errorStream = ssh2_fetch_stream($stream, SSH2_STREAM_STDERR);
+        stream_set_blocking($errorStream, TRUE);
+        stream_set_blocking($stream, TRUE);
+        $output = stream_get_contents($stream);
+        $error_output = stream_get_contents($errorStream);
+        fclose($stream);
+        fclose($errorStream);
+        if (!empty($error_output)) {
+            throw new ConnectionException($error_output, $this->profile);
```

Comment:
ok to throw an exception if command fail, but not a ConnectionException. Maybe use deleted Exception, or create CommandException